inspirehep/dojson/hepnames/fields/bd1xx.py

```diff
@@ -263,65 +263,58 @@ def hidden_notes2marc(self, key, value):
 
 @hepnames.over('positions', '^371..')
 @utils.for_each_value
-@utils.filter_values
 def positions(self, key, value):
-    """Position information.
-
-    In dates field you can put months in addition to years. In this case you
-    have to follow the convention `mth-year`. For example: `10-2012`.
-    """
-    curated_relation = False
+    """Positions that an author held during their career."""
+    curated = False
+    current = False
     recid = None
-    status = ''
-    recid_status = force_force_list(value.get('z'))
-    for val in recid_status:
-        if val.lower() == 'current':
-            status = val
+
+    recid_or_status = force_force_list(value.get('z'))
+    for el in recid_or_status:
+        if el.lower() == 'current':
+            current = True
         else:
-            try:
-                recid = int(val)
-                curated_relation = True
-            except ValueError:
-                pass
+            curated = el.isdigit()
+            if curated:
```

Comment:
This reads a bit weird. It took me a while: it's not because it's curated that you can assign it. But finally it's correct :)